### PR TITLE
[HOTFIX][BUILD] Do not use contains in Option in JdbcRelationProvider

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes the fix the use of `contains` API which only exists from Scala 2.11.
## How was this patch tested?

Manually checked:

``` scala
scala> val o: Option[Boolean] = None
o: Option[Boolean] = None

scala> o == Some(false)
res17: Boolean = false

scala> val o: Option[Boolean] = Some(true)
o: Option[Boolean] = Some(true)

scala> o == Some(false)
res18: Boolean = false

scala> val o: Option[Boolean] = Some(false)
o: Option[Boolean] = Some(false)

scala> o == Some(false)
res19: Boolean = true
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@yhuai @zsxwing Do you mind if I ask to take a look please?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ class JdbcRelationProvider extends CreatableRelationProvider
       if (tableExists) {
         mode match {
           case SaveMode.Overwrite =>
-            if (isTruncate && isCascadingTruncateTable(url).contains(false)) {
+            if (isTruncate && isCascadingTruncateTable(url).exists(_ == false)) {
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala]: The current one also looks good to me. :)

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala]: Oh, yes I think that looks nicer. Thanks.

**Review by srowen — APPROVED** [review]
+1 yeah been bitten by this one before

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala]: Thanks!

**Comment by gatorsmile** [comment]
LGTM pending tests

**Comment by gatorsmile** [comment]
Actually, I hit this before. Sorry, I did not catch it when I merging the PR. Next time, I will be more careful about it.

**Comment by HyukjinKwon** [comment]
I apologize and will definitely be careful as well.

**Comment by dongjoon-hyun** [comment]
BTW, could you update the PR description, @HyukjinKwon ? Maybe, removing `exists` part?

**Comment by HyukjinKwon** [comment]
@dongjoon-hyun I already did :)

**Comment by dongjoon-hyun** [comment]
Yep. I knew. What I'm asking is we don't need to explain 'contains' usage here in this PR.
I prefer not to advertise `contains` for `Option[Boolean]`.

```
scala> val o: Option[Boolean] = None
o: Option[Boolean] = None

scala> o.contains(false)
res17: Boolean = false

scala> val o: Option[Boolean] = Some(true)
o: Option[Boolean] = Some(true)

scala> o.contains(false)
res18: Boolean = false

scala> val o: Option[Boolean] = Some(false)
o: Option[Boolean] = Some(false)

scala> o.contains(false)
res19: Boolean = true
```

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
test this please

**Comment by zsxwing** [comment]
retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala (hunk 1 above)
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcRelationProvider.scala]: Yes - same thing as contains, don't use exists as well.

**Comment by rxin** [comment]
Merging in master.

**Pull request closed without merging**